tests/test_disjoint.ml

```diff
@@ -0,0 +1,31 @@
+module Atomic = Dscheck.TracedAtomic
+
+let two_vars () =
+  let a = Atomic.make 0 in
+  let b = Atomic.make 0 in
+  let c = Atomic.make 0 in
+  let d = Atomic.make 0 in
+
+  Atomic.trace (fun () ->
+      Atomic.spawn (fun () ->
+          Atomic.set a 1;
+          Atomic.set b 1;
+          Atomic.set c 1;
+          Atomic.set d 1)
+      |> ignore;
+
+      Atomic.spawn (fun () ->
+          Atomic.set d 2;
+          Atomic.set c 2;
+          Atomic.set b 2;
+          Atomic.set a 2)
+      |> ignore;
+
+      Atomic.final (fun () -> ()));
```

Comment:
It should help to insert some debug print here:

```ocaml
Atomic.final (fun () ->
  Format.printf "a=%i b=%i c=%i d=%i@."
    (Atomic.get a) (Atomic.get b) (Atomic.get c) (Atomic.get d)));
```
and investigate why it outputs these terminal states:
```
a=2 b=2 c=2 d=2
a=2 b=2 c=2 d=2
a=2 b=2 c=2 d=1
a=2 b=2 c=2 d=2
a=2 b=2 c=2 d=1
```
rather than:
```
a=2 b=2 c=2 d=2
a=2 b=2 c=2 d=1
a=2 b=2 c=1 d=1
a=2 b=1 c=1 d=1
a=1 b=1 c=1 d=1
```